lib/core/sdam/server_description.js

```diff
@@ -143,6 +145,29 @@ class ServerDescription {
       topologyVersionsEqual
     );
   }
+
+  /**
+   * Creates a new ServerDescription from this
+   * serverDescription with no ismaster response.
+   *
+   * @returns ServerDescription
+   */
+  reset() {
```

Comment:
can we rename this to something that indicates the immutability of the action? This method name indicates action on the `ServerDescription` instance itself (mutating internal state), when its really a factory method to return a clone of this instance with its state transitioned to `Unknown`. Perhaps a better name would be `toUnknown` or something similar?